### PR TITLE
Deprecate FileLink.process

### PR DESCRIPTION
**Pull request by kroenlein**
# Citrine Python PR

## Description 
With the current ingest workflows, this FileLink.process workflow is no longer relevant.  Deprecating the interface here with the expectation of the removal of routes from the Platform.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [X] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [X] I have bumped the version in __version\__.py

### CODE REVIEW

**Comment by kroenlein** [comment]
E2Es with these updates: https://jenkins.corp.citrine.io/job/nextgen-ITs/35937

**Comment by anoto-moniz** [comment]
LGTM, I'd just like us to point the user of the deprecated methods in the right direction, if possible.

File: src/citrine/resources/file_link.py [hunk 1]
```diff
@@ -718,6 +719,9 @@ def read(self, *, file_link: Union[str, UUID, FileLink]):
         download_response = requests.get(final_url)
         return download_response.content
 
+    @deprecated(deprecated_in="2.4.0",
+                removed_in="3.0.0",
+                details="The process file protocol is deprecated in favor of other ingest routes")
```
[the remark this answers is not in the document]
Comment (kroenlein) [reply src/citrine/resources/file_link.py:724]: No, because Scala ingest is not exposed through the SDK.

[the remark this answers is not in the document]
Comment (anoto-moniz) [reply src/citrine/resources/file_link.py:724]: Why can't we point them towards the UI? It's not ideal, sure, but I think it's better to give them some direction.

**Review by anoto-moniz — APPROVED** [review]

**Review by anoto-moniz — APPROVED** [review]

**Pull request merged**